qubesbuilder/executors/container.py

```diff
@@ -72,14 +84,15 @@ def __init__(
             try:
                 # Check if we have the image locally
                 docker_image = client.images.get(image)
-            except (PodmanError, DockerException) as e:
+            except (PodmanError, DockerException):
                 # Try to pull the image
                 try:
                     docker_image = client.images.pull(image)
                 except (PodmanError, DockerException) as e:
                     raise ExecutorError(f"Cannot find {image}.") from e
 
-        self.attrs = docker_image.attrs
+        self._attrs = docker_image.attrs
+        self._container = None
```

Comment:
This could use a comment (at least in a commit message) why it's moved here (IIUC for easier testing?). Otherwise this change looks unwanted, as lifetime of the container is bound to the `run()` method, and moving it outside forced you to add a bunch of checks that weren't necessary before.